purple-pill/js/controllers/fontexportController.js

```diff
@@ -32,140 +112,107 @@ function($scope, $http, sharedScope, $timeout) {
+            return percentage;
         }
 
-        function exportFont_compute_glyphs(){
+        var index = 0
+          , totalInstances = exportObjects.length
+          , obj = exportObjects.pop()
+          ;
+        function exportFontComputeGlyphs(){
             var text
-              , current_glyph
-              , total_glyphs
-              , glyph_name
-              , it = generator.next()
+              , percentage
+              , zippedData
+              , it = obj.generator.next()
               ;
             if (!it.done){
-                current_glyph = it.value['current_glyph'];
-                total_glyphs = it.value['total_glyphs'];
-                glyph_id = it.value['glyph_id'];
-                text = total_instances + " instances to export. Calculating glyph '" + glyph_id + "' (" + (current_glyph+1) + " of " + total_glyphs + ") of instance #" + (current_instance+1);
-                setProgress(glyphs_phase_percentage * (current_instance + ((current_glyph+1) / total_glyphs))/total_instances, text);
-                $timeout(exportFont_compute_glyphs, UI_UPDATE_TIMESLICE);
+                text = exportingGlyphMessage(it, index);
+                percentage = calculateGlyphsProgress(it, index, totalInstances);
+                progress.set(percentage, text);
+                $timeout(exportFontComputeGlyphs, UI_UPDATE_TIMESLICE);
             } else {
-                delete generator;
-                if (++current_instance < total_instances) {
-                    generator = getExportGenerator(current_instance);
-                    $timeout(exportFont_compute_glyphs, UI_UPDATE_TIMESLICE);
+                index++;
+                delete obj.generator;
+
+                if (obj.fileFormat == "UFO"){
+                    zippedData = $scope.data.stateful.project.getZipFromIo(false, obj.io, obj.getFileName(), "uint8array");
+                    bundleFolder.file(obj.getFileName()+".zip", zippedData, {binary:true});
+                    delete zippedData;
+                } else { /* obj.fileFormat == "OTF" */
+                    var plainData = obj.io.readFile(false, obj.getFileName());
+                    bundleFolder.file(obj.getFileName(), plainData, {binary:true});
+                }
+                delete obj;
+
+                if (exportObjects.length) {
+                    obj = exportObjects.pop();
+                    $timeout(exportFontComputeGlyphs, UI_UPDATE_TIMESLICE);
                 } else {
-                    current_instance = 0;
-                    text = total_instances + " instances to export in .ufo.zip format, zipping instance #" + (current_instance+1) + " (can take a while)";
-                    setProgress(glyphs_phase_percentage, text);
-                    $timeout(exportFont_pack_instance_chunk, UI_UPDATE_TIMESLICE);
+                    bundleData = bundle.generate({type:"blob"});
+                    setDownloadBlobLink(bundleFileName, bundleData, bundleFileName);
```

Comment:
instead of the the two lines
```
setDownloadBlobLink(bundleFileName, bundleData, bundleFileName);
 exportIsRunning = false;
```

You should do:
```
callback(bundleData);
```

if using a promise:
```
promise.resolve(bundleData)
```

then, your callback executes these two lines.